crates/solver/src/solver/http_solver/settlement.rs

```diff
@@ -100,19 +110,24 @@ impl IntermediateSettlement {
             .collect();
         let executions = merge_and_order_executions(executions_amm, executions_interactions)?;
         let prices = match_settled_prices(executed_limit_orders.as_slice(), settled.prices)?;
+        let approvals = compute_approvals(allowance_manager, settled.approvals).await?;
         Ok(Self {
             executed_limit_orders,
             executions,
             prices,
+            approvals,
         })
     }
 
     fn into_settlement(self) -> Result<Settlement> {
         let mut settlement = Settlement::new(self.prices);
-        for order in self.executed_limit_orders.iter() {
+        for order in self.executed_limit_orders {
             settlement.with_liquidity(&order.order, order.executed_amount())?;
         }
-        for execution in self.executions.iter() {
+        for approval in self.approvals {
+            settlement.encoder.append_to_execution_plan(approval);
+        }
```

Comment:
I intentionally add them before `self.executions` which are already orders by their execution plan.

They need to come first in order for the approvals to be setup for the AMM swaps to work.

I will add a comment clarifying this.

Reply:
I was aware that the approval should probably happen before the executions. I was just not sure whether it would make a difference going in the order of:
1. approvals
2. user orders
3. executions

But I guess not.

Reply:
Ah, you mean using pre-interactions. That shouldn't be necessary at the moment. As long as they come before any other interactions, it should work.

Reply:
Added comment clarifying.